test/extensions/filters/http/common/empty_http_filter_config.h

```diff
@@ -44,6 +44,29 @@ class EmptyHttpFilterConfig : public Server::Configuration::NamedHttpFilterConfi
   const std::string name_;
 };
 
+class UpstreamFilterConfig : public virtual Server::Configuration::UpstreamHttpFilterConfigFactory {
```

Comment:
I don't think this needs virtual inheritance, can you clarify where you ended up needing it?